.github/actions/build_apko/action.yml

```diff
@@ -0,0 +1,61 @@
+name: Build apko image
+description: Build an apko image based on a supplied .yaml file
+
+inputs:
+  apkoConfig:
+    description: "Path to the apko .yaml config file. If left empty, all images will be built."
+    required: false
+  registry:
+    description: "Container registry to use"
+    default: "ghcr.io"
+    required: true
+  githubToken:
+    description: "GitHub authorization token"
+    required: true
+
+# Linux runner only (podman required)
+runs:
+  using: "composite"
+  steps:
+    - name: Install podman
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y podman
+
+    - name: Log in to the Container registry
+      id: podman-login
+      uses: redhat-actions/podman-login@v1
+      with:
+        registry: ${{ inputs.registry }}
+        username: ${{ github.actor }}
+        password: ${{ inputs.githubToken }}
+
+    - name: Build apko images
+      shell: bash
+      run: |
+        if [ -z "${{ inputs.apkoConfig }}" ]; then
+          echo "Building all images in image"
+          for imageConfig in image/apko/*.yaml; do
+            echo "Building image for $imageConfig"
+
+            imageName=$(basename $imageConfig | cut -d. -f1 )
+            registry=${{ inputs.registry }}/edgelesssys/apko-${imageName}
+            outTar=$imageName.tar
+
+            podman run -v "$PWD":/work cgr.dev/chainguard/apko build $imageConfig $registry $outTar
+            podman load < $outTar
+            podman push $registry
+
+          done
+        else
+          echo "Building image for ${{ inputs.apkoConfig }}"
+
+          imageName=$(basename ${{ inputs.apkoConfig }} | cut -d. -f1 )
+          registry=${{ inputs.registry }}/edgelesssys/apko-${imageName}
+          outTar=$imageName.tar
+
+          podman run -v "$PWD":/work cgr.dev/chainguard/apko build ${{ inputs.apkoConfig }} $registry $outTar
+          podman load < $outTar
+          podman push $registry
```

Comment:
We should also use cosign for the `apko` generated images.

`libcryptsetup` will be the final image, right? So users should be able to check it.

In case where we use `apko` images as a base image for further steps, we should check the signature before proceeding to use it.

Reply:
Currently, `libcryptsetup` is only part of the [alpine-libcryptsetup.yaml](./image/apko/alpine-libcryptsetup.yaml) image that can be used as a base image for an `apko` container, if the application running in this container needs this libraray.

> I was trying to motivate adding cosign signatures to the `apko` images so that "users of the libcryptsetup image can check the signature on it via cosign verify".

I'm not sure if I understand what you mean. Which signatures should be stored in which images and in when should these signatures be verified?